src/Language/Haskell/Refact/Utils/MonadFunctions.hs

```diff
@@ -428,6 +439,11 @@ logParsedSource str = do
 
 -- ---------------------------------------------------------------------
 
+logAst :: (SYB.Data a) => String -> a -> RefactGhc ()
+logAst str a = logm $ str ++ "\n[" ++ (SYB.showData SYB.Parser 3 a) ++ "]"
```

Comment:
I generally use `logDataWithAnns` which is the same as `showData` but includes the current annotations. FYI. It may be that this is not needed.